tt_metal/hw/firmware/src/brisc.cc

```diff
@@ -76,6 +76,21 @@ void enable_power_management() {
     // Mask and Hyst taken from tb_tensix math_tests
     uint32_t pm_mask = 0xFFFF;
     uint32_t pm_hyst = 32;
+
+#ifdef ARCH_BLACKHOLE
+    {
+        uint32_t hyst0_reg_data = ((pm_hyst) << 24) | ((pm_hyst) << 16) | ((pm_hyst) << 8) | pm_hyst;
+        uint32_t hyst1_reg_data = ((pm_hyst) << 24) | ((pm_hyst) << 16) | ((pm_hyst) << 8) | pm_hyst;
+        uint32_t hyst2_reg_data = ((pm_hyst) << 24) | ((pm_hyst) << 16) | ((pm_hyst) << 8) | pm_hyst;
+
+        *((uint32_t volatile*)RISCV_DEBUG_REG_CG_CTRL_HYST0) = hyst0_reg_data;
+        *((uint32_t volatile*)RISCV_DEBUG_REG_CG_CTRL_HYST1) = hyst1_reg_data;
+        *((uint32_t volatile*)RISCV_DEBUG_REG_CG_CTRL_HYST2) = hyst2_reg_data;
+    }
+    /*FIXME: need to deal with srcb ctrl bit not fitting in 16 bits. For  */
+    /*now just always turn it on */
+    *((uint32_t volatile*)RISCV_DEBUG_REG_CG_CTRL_EN) = 0x10000 | (pm_mask);
```

Comment:
it looks like the only diff between BH and other archs is using `hyst_val` vs `pm_hyst` in `hyst*_reg_data` and BH writes to `RISCV_DEBUG_REG_CG_CTRL_EN` vs other archs do `core.ex_setc16(CG_CTRL_EN_Regblocks_ADDR32, pm_mask, instrn_buf[0]);`

Is there a way to consolidate the common pieces?

Reply:
Done, consolidated the common parts.